### DISCUSSION TITLE
在 zk-SNARKs 中为什么可信设置只生成了一次随机挑战数s就可以反复在证明过程中使用

### DISCUSSION THREAD

**Discussion by eiTT-01** [zkp-co-learning/zkp-co-learn#67] [Q&A（提问题在隔壁~）] [answered]
在零知识证明过程中，验证者需要对证明者发起多次挑战，来确保证明的正确性。这也是在多项式方程中验证不同解的过程。
但在非交互式 zk-SNARKs 中，挑战随机数 s 是事先由可信第三方固定生成的，并不是在每次验证时都重新生成的。那么，验证者如何对不同的证明者发起多次挑战呢？

**Comment by lkiversonlk** [comment, marked as answer]
你指的是复用同一个证明材料？

我觉得你可以参考对付延展性攻击的方法，思路应该是一样的。

https://mp.weixin.qq.com/s/ge3QJOdIqrDiRMBgh6BBeg
